sale_order_invoice_amount/models/sale_order.py

```diff
@@ -42,6 +42,13 @@ def _compute_invoice_amount(self):
                                 invoice.company_id,
                                 invoice.invoice_date or fields.Date.today(),
                             )
+                        elif (
```

Comment:
Hy,

Thanks for the proposal.
I totally agree on the case. 
Yet I think it should be great to simplify. Actually this line L37 is the same as your addition.
So in the end we have 2 main cases and not 3 : 
* One case when `rec.currency_id == invoice.company_currency_id` fair case in L53
* the other one when `rec.currency_id != invoice.company_currency_id` andIMHO, we should always use the convert method to prevent the enumeration of cases and improve readability.

Of course this is my humble opinion